### PR TITLE
Suppress Jmap Integration test logs

### PR DESCRIPTION
**Pull request by trantienduchn**
Addressed Item:
- RabbitMQ queue deletions should be executed in PreDestroy phrase
- ~ES reindex UpperCase Alias name error~
- ~SpamAssassin connecting error logs~

Result of RabbitMQFilterTest:
Before: 13000++ lines of log
After: 7600 lines of logs

### CODE REVIEW

File: mailbox/plugin/quota-search-elasticsearch/src/main/java/org/apache/james/quota/search/elasticsearch/QuotaRatioElasticSearchConstants.java [hunk 1]
```diff
@@ -30,8 +30,8 @@
         String QUOTA_RATIO = "quotaRatio";
     }
 
-    WriteAliasName DEFAULT_QUOTA_RATIO_WRITE_ALIAS = new WriteAliasName("quotaRatioWriteAlias");
-    ReadAliasName DEFAULT_QUOTA_RATIO_READ_ALIAS = new ReadAliasName("quotaRatioReadAlias");
+    WriteAliasName DEFAULT_QUOTA_RATIO_WRITE_ALIAS = new WriteAliasName("quota_ratio_write_alias");
```
[the remark this answers is not in the document]
Comment (trantienduchn) [reply mailbox/plugin/quota-search-elasticsearch/src/main/java/org/apache/james/quota/search/elasticsearch/QuotaRatioElasticSearchConstants.java]: It's my proposal, and still open for evaluation from people have better knowledge about the impact of this change. Seem ES reject to index alias:
```
[org.apache.james.quota.search.elasticsearch.events.ElasticSearchQuotaMailboxListener]- Can not index quota ratio
org.elasticsearch.indices.InvalidIndexNameException: Invalid index name [quotaRatioWriteAlias], must be lowercase
```
Then this problem should be critical.

File: server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing/src/test/resources/logback-test.xml [hunk 2]
```diff
@@ -1,6 +1,13 @@
 <configuration>
 
     <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+        <filter class="ch.qos.logback.core.filter.EvaluatorFilter">
```
[the remark this answers is not in the document]
Comment (trantienduchn) [reply server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing/src/test/resources/logback-test.xml]: The problem is, in a integration test maven module, we use same `mailetcontainer.xml` for all integration tests including test don't use SpamAssassin and SpamAssasinContractTest. If I remove SpamAssasin maillet from xml file, then I cannot test `SpamAssasinContractTest`. Possible solution I could imagine is this way or modify `CamelMailetContainerModule` to be able to bind a custom `MailetModuleConfigurationPerformer` in test

File: mailbox/plugin/quota-search-elasticsearch/src/main/java/org/apache/james/quota/search/elasticsearch/QuotaRatioElasticSearchConstants.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (aduprat) [reply mailbox/plugin/quota-search-elasticsearch/src/main/java/org/apache/james/quota/search/elasticsearch/QuotaRatioElasticSearchConstants.java]: Why we don't have such logs in production?
Maybe it's an EmbeddedElasticSearch issue...

[the remark this answers is not in the document]
Comment (trantienduchn) [reply mailbox/plugin/quota-search-elasticsearch/src/main/java/org/apache/james/quota/search/elasticsearch/QuotaRatioElasticSearchConstants.java]: I don't think so, [see](https://www.elastic.co/guide/en/elasticsearch/reference/6.4/indices-create-index.html)
Maybe alias name in production is using another value by configuring this properity `elasticsearch.alias.write.quota.ratio.name` in `elasticsearch.properities`

File: server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing/src/test/resources/logback-test.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mbaechler) [reply server/protocols/jmap-integration-testing/cassandra-jmap-integration-testing/src/test/resources/logback-test.xml]: Well, we need to fix the root cause: we use config files in tests and are it's painful to manage we share these files between as many tests as possible.
I already argued in the past that we should be able to start a James server with default behavior without any config file and customize everything with a nice DSL.
Anyway, let's not put a carpet on top of the mess, we should rather clean the mess.

File: mailbox/plugin/quota-search-elasticsearch/src/main/java/org/apache/james/quota/search/elasticsearch/QuotaRatioElasticSearchConstants.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (aduprat) [reply mailbox/plugin/quota-search-elasticsearch/src/main/java/org/apache/james/quota/search/elasticsearch/QuotaRatioElasticSearchConstants.java]: We are using ES 2.4.6, which has no such limitation:
https://www.elastic.co/guide/en/elasticsearch/reference/2.4/indices-create-index.html

Here are the indices created by the docker-compose:
```
✗ curl http://192.168.32.5:9200/_aliases\?pretty\=1
{
  "mailbox_v1" : {
    "aliases" : {
      "mailboxReadAlias" : { },
      "mailboxWriteAlias" : { }
    }
  },
  "quota_ratio_v1" : {
    "aliases" : {
      "quotaRatioReadAlias" : { }
    }
  }
}
```

**Comment by hoangdat** [comment]
Build triggered. sha1 is original commit.

**Comment by hoangdat** [comment]
Build started sha1 is original commit.

**Comment by hoangdat** [comment]
Build finished. No test results found.

**Comment by hoangdat** [comment]
Build triggered. sha1 is original commit.

**Comment by hoangdat** [comment]
Build started sha1 is original commit.

**Comment by trantienduchn** [comment]
test this please

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/MessageBatcher.java [hunk 3]
```diff
@@ -19,12 +19,13 @@
 
 package org.apache.james.mailbox.store;
 
-import java.util.ArrayList;
```
Comment (chibenwa) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/MessageBatcher.java]: Changeset not related?

**Comment by trantienduchn** [comment]
```
[799047bb906c5cbd00d64c2a50116d7510be141f] [ERROR] Errors: 
[799047bb906c5cbd00d64c2a50116d7510be141f] [ERROR]   SendMDNMethodTest.sendMDNShouldSendAMDNBackToTheOriginalMessageAuthor » RejectedExecution
[799047bb906c5cbd00d64c2a50116d7510be141f] [INFO] 
[799047bb906c5cbd00d64c2a50116d7510be141f] [ERROR] Tests run: 89, Failures: 0, Errors: 1, Skipped: 0
```
test this please

**Comment by trantienduchn** [comment]
> I don't understand you commit.
We previously leverage CleanupTasksPerformer.CleanupTask to register our cleanup task.
Now, we rely on @Predestroy instead ?
What is the difference ?

The difference is, `CleanupTask` is performed before PreDestroy stage. Then all queues are deleted but at that moment `RemoteDelivery` is still dequeueing => full of rabbitmq channel closing logs printed, and dequeuing is only stopped in `RemoteDelivery.shutDown()` at Predestroy stage . By my proposal, `RemoteDelivery.shutDown()` is performed first, and then delete all mailqueues is performed at the last.

**Comment by chibenwa** [comment]
Maybe the solution is to stop the Remote Delivery stuff through a cleanUp task & assign a priority to CleanUp tasks.

Matthieu, the issue is that CleanUp tasks are called first, then @PreDestroy stages . We have to re-think the way we stop James to handle that in a non hacky way.

**Comment by mbaechler** [comment]
See https://github.com/linagora/james-project/pull/2022 for a possible implementation

**Comment by trantienduchn** [comment]
see #2022

**Pull request closed without merging**